application/src/main/java/org/opentripplanner/routing/algorithm/mapping/RaptorPathToItineraryMapper.java

```diff
@@ -361,15 +360,24 @@ private List<Leg> mapNonTransitLeg(
           .build()
       );
     } else {
-      var legTransferSearchRequest = transferStreetRequest
-        .copyOf(createZonedDateTime(pathLeg.fromTime()).toInstant())
-        .build();
-      var initialStates = State.getInitialStates(
-        Set.of(edges.getFirst().getFromVertex()),
-        legTransferSearchRequest
-      );
-      var state = EdgeTraverser.traverseEdges(initialStates, edges);
-      var graphPath = new GraphPath<>(state.get());
+      StateEditor se = new StateEditor(edges.get(0).getFromVertex(), transferStreetRequest);
+      se.setTimeSeconds(createZonedDateTime(pathLeg.fromTime()).toEpochSecond());
+
+      State s = se.makeState();
+      ArrayList<State> transferStates = new ArrayList<>();
+      transferStates.add(s);
+      for (Edge e : edges) {
+        var states = e.traverse(s);
+        if (State.isEmpty(states)) {
+          s = null;
+        } else {
+          transferStates.add(states[0]);
+          s = states[0];
+        }
```

Comment:
We can revert this for the performance benefit. It may be useful to add a check here to avoid problems related to A* searches returning multiple states similar to `EdgeTraverser`.

Reply:
For reference, flame graph showing the impact of #6238 on RaptorRequestTransferCache

![Screenshot from 2025-01-07 11-02-57](https://github.com/user-attachments/assets/6783c98a-ed25-4f11-a224-fc2c8a188369)